torch/distributed/_sharded_tensor/api.py

```diff
@@ -71,50 +82,55 @@ class ShardedTensorMetadata(object):
     # Size of each dim of the overall Tensor.
     size: torch.Size = field(default=torch.Size([]))
 
-    # Regular tensor fields
-    dtype: torch.dtype = field(default=torch.get_default_dtype())
-    layout: torch.layout = field(default=torch.strided)
-    requires_grad: bool = False
-    memory_format: torch.memory_format = field(default=torch.contiguous_format)
-    pin_memory: bool = False
+    tensor_properties: TensorProperties = field(
```

Comment:
Can we still preserve the old constructor? i.e. User can still do things like this:

```
sharded_tensor_metadata = ShardedTensorMetadata(
    shards_metadata = shards_metadata,
    size = torch.Size([10, 10]),
    dtype = torch.get_default_dtype(),
    layout = torch.strided,
    requires_grad=False,
    ...
)
```
We can construct the TensorProperties instance based on user input inside ShardedTensorMetadata constructor without letting user know there's a `TensorProperties` type, that would keep the old API to user and still help us merge the common fields.

Reply:
It seems cleaner to use TensorProperties here though, we can just update the callpaths to use this instead since this is not yet a publicly released feature?

Reply:
Yeah it's cleaner from code perspective, the only thing I am worrying is that, this creates a two-level abstraction for ShardedTensorMetadata, user now need to know there're two data structures need to be built (TensorProperties -> ShardedTensorMetadata), which might hurt UX. It's unclear to me from a UX perspective though, so I don't have a strong opinion on this, we can keep it this way.

Reply:
Thanks Wanchao and Pritam.

Wanchao and I had a quick offline chat and agreed to keep TensorProperties as is. 

Context: 
a) _sharded_tensor is still experimental, we need a few more discussions to finalize the API
b) now there is only one caller (pyper checkpoint_utils.py) -- updated the usage on D30472281
c) we need to update the caller side again once the API is locked down. 

We will continue to polish the API and publish it once everything is stabilized.